src/base/matrix.rs

```diff
@@ -360,6 +360,29 @@ impl<N: Scalar, R: Dim, C: Dim, S: Storage<N, R, C>> Matrix<N, R, C, S> {
         res
     }
 
+    /// Returns a matrix containing the result of `f` applied to each of its entries. Unlike `map`,
+    /// `f` also gets passed the row and column index, i.e. `f(value, row, col)`.
+    #[inline]
+    pub fn map_with_location<N2: Scalar, F: FnMut(N, usize, usize) -> N2>(&self, mut f: F) -> MatrixMN<N2, R, C>
```

Comment:
The argument order should be `FnMut(usize, usize, N)` so this is more in line with the rest of the API that generally takes indices as the first arguments, and more similar to what would give `.enumerate().map(f)`.